### ISSUE TITLE
Y.js V2 Update Format

### ISSUE THREAD

**Issue by MentalGear** [partykit/partykit#808]
(I didn't find anything in docs on this.)

Please support Y.js V2 Update Format, which provides much better compression.
https://github.com/yjs/yjs#using-v2-update-format

**Comment by threepointone** [comment]
Yeah we should do this, but I don't think it'll happen soon; we'd have to make sure it' backward compatible with existing projects/persisted data. Happy to review a PR.

**Comment by MentalGear** [comment]
How's about making it a opt-in, so it's easier to get started? New projects could start with V2, and get the performance benefits, right away!

**Comment by MentalGear** [comment]
@threepointone  What do you think about the previous idea mentioned? If you could outline how this first opt-in version can be added, I'm open to give it a try.

**Comment by raymonddaikon** [comment]
I think just adding an option to YPartyKitProvider would work. So something like `{ updateProtocol: 'V1' }` then using that as a flag to either transform the  incoming update via the helper function `convertUpdateFormatV1ToV2` or just directly pass the update.

**Comment by Ryner01** [comment]
Hello, wanted to bump this issue and ask about updates. Is it now possible to use V2 protocol? 

Some of our yjs docs are pretty big and this would help a lot.
